components/Profile/PointsShowcase.jsx

```diff
@@ -1,151 +1,32 @@
-import { Card, Col, Row, Spin, Typography } from 'antd';
+import { Col, Row, Typography } from 'antd';
 import Link from 'next/link';
-import Script from 'next/script';
 import PropTypes from 'prop-types';
-import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { memo } from 'react';
 import styled from 'styled-components';
 
+import { TweetShape } from 'common-util/prop-types';
+
+import { TweetEmbed } from './TweetEmbed';
 import { shuffleArray } from './utils';
 
 const MAX_TWEETS_SHOWN = 2;
 
 const { Paragraph, Text } = Typography;
 
-const TweetLoader = styled.div`
-  display: flex;
-  justify-content: center;
-  margin: 16px;
-`;
-
 const StyledCol = styled(Col)`
   flex: auto;
   max-width: 306px;
 `;
-
-const StyledCard = styled(Card)`
-  max-width: 282px;
-  width: 100%;
-`;
-
 const NoTweetsText = styled(Paragraph)`
   max-width: 400px;
 `;
 
-const TweetEmbed = ({ points, tweetId, isScriptReady, isError, onError }) => {
-  const isLoaded = useRef(false);
-  const [isLoading, setIsLoading] = useState(true);
-
-  const createTweet = useCallback(() => {
-    try {
-      if (isLoaded.current) return;
-
-      const element = document.getElementById(`tweet-container-${tweetId}`);
-      if (!element) return;
+export const PointsShowcase = memo(({ tweetsData }) => {
```

Comment:
oh you're right I forgot to pass equal props condition, will do
it's useful here not only because it should (as I was planning to do) check the tweetsData update, but also not be re-rendered when parent's component states change, which will cause this to re-render and tweets to shuffle again and embed another set of tweets